glotaran/parameter/test/test_parameter.py

```diff
@@ -1,169 +1,150 @@
-    assert len(list(params.all())) == 3
-    assert [p.label for _, p in params.all()] == [f"{i}" for i in range(5, 2, -1)]
-    assert [p.value for _, p in params.all()] == list(range(1, 4))
+    parameters = [Parameter.from_list(v) for v in params]
 
+    assert parameters[0].value == 1.0
+    assert not parameters[0].non_negative
+    assert parameters[0].minimum == -1
+    assert parameters[0].maximum == 1
+    assert not parameters[0].vary
 
-def test_parameter_options():
-    params = """
-    - ["5", 1, {non-negative: false, min: -1, max: 1, vary: false}]
-    - ["6", 4e2, {non-negative: true, min: -7e2, max: 8e2, vary: true}]
-    - ["7", 2e4]
-    """
-
-    params = load_parameters(params, format_name="yml_str")
-
-    assert params.get("5").value == 1.0
-    assert not params.get("5").non_negative
-    assert params.get("5").minimum == -1
-    assert params.get("5").maximum == 1
-    assert not params.get("5").vary
-
-    assert params.get("6").value == 4e2
-    assert params.get("6").non_negative
-    assert params.get("6").minimum == -7e2
-    assert params.get("6").maximum == 8e2
-    assert params.get("6").vary
-
-    assert params.get("7").value == 2e4
-    assert not params.get("7").non_negative
-    assert params.get("7").minimum == float("-inf")
-    assert params.get("7").maximum == float("inf")
-    assert params.get("7").vary
-
-
-def test_parameter_block_options():
-    params = """
-    block:
-        - 1.0
-        - [3.4, {vary: true}]
-        - {vary: false}
-    """
-
-    params = load_parameters(params, format_name="yml_str")
-    assert not params.get("block.1").vary
-    assert params.get("block.2").vary
-
-
-def test_parameter_set_from_group():
-    """Parameter extracted from group has correct values"""
-    group = load_parameters(
-        "foo:\n  - [\"1\", 123,{non-negative: true, min: 10, max: 8e2, vary: true, expr:'2'}]",
-        format_name="yml_str",
-    )
-    parameter = Parameter(full_label="foo.1")
-    parameter.set_from_group(group=group)
+    assert parameters[1].value == 4e2
+    assert parameters[1].non_negative
+    assert parameters[1].minimum == -7e2
+    assert parameters[1].maximum == 8e2
+    assert parameters[1].vary
 
-    assert parameter.value == 123
-    assert parameter.non_negative is True
-    assert np.allclose(parameter.minimum, 10)
-    assert np.allclose(parameter.maximum, 800)
-    assert parameter.vary is True
-    # Set to None since value and expr were provided?
-    assert parameter.expression is None
+    assert parameters[2].value == 2e4
+    assert not parameters[2].non_negative
+    assert parameters[2].minimum == float("-inf")
+    assert parameters[2].maximum == float("inf")
+    assert parameters[2].vary
 
 
 def test_parameter_value_not_numeric_error():
     """Error if value isn't numeric."""
-    with pytest.raises(TypeError, match="Parameter value must be numeric"):
-        Parameter(value="foo")  # type:ignore[arg-type]
+    with pytest.raises(TypeError):
```

Comment:
Please add back the error message matching.

Reply:
see below